### PR TITLE
:sparkles: `[environment]` Manipulate environment variable collections

### PR DESCRIPTION
**Pull request by acabarbaye**
### Description
- `[platform]` Define a portable way to do parameter substitution
- `[environment]` Add utilities to manipulate environment variables such as merges, sorting, ensure uniqueness

### Test Coverage

- [x]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: utils/platform/os.go [hunk 1]
```diff
@@ -179,6 +179,44 @@ func GetRAM() (ram RAM, err error) {
...
+// - the first element is the parameter to substitute
+// - if find and replace is also wanted, pass the pattern and the replacement as following arguments in that order.
+func SubstituteParameterUnix(parameter ...string) string {
+	if len(parameter) < 1 {
+		return "${}"
+	}
+	if len(parameter) < 3 || parameter[1] == "" {
+		return fmt.Sprintf("${%v}", parameter[0])
+	}
+	return fmt.Sprintf("${%v//%v/%v}", parameter[0], parameter[1], parameter[2])
```
Comment (joshjennings98) [inline utils/platform/os.go:203]: what if the parameters contain "/", is that possible

**Review by joshjennings98 — APPROVED** [review]
Just one comment but if that is covered then it looks all good

File: utils/platform/os.go (hunk 1 above)
Comment (acabarbaye) [reply utils/platform/os.go:203]: this is fair enough. I shall add a validation rule

Comment (acabarbaye) [reply utils/platform/os.go:240]: Added

**Pull request merged**